lib/internal/modules/esm/module_job.js

```diff
@@ -151,7 +167,33 @@ class ModuleJob {
     await this.instantiate();
     const timeout = -1;
     const breakOnSigint = false;
-    await this.module.evaluate(timeout, breakOnSigint);
+    try {
+      await this.module.evaluate(timeout, breakOnSigint);
+    } catch (e) {
+      if (e?.name === 'ReferenceError' &&
+          isCommonJSGlobalLikeNotDefinedError(e.message)) {
+        e.message += ' in ES module scope';
+
+        if (StringPrototypeStartsWith(e.message, 'require ')) {
+          e.message += ', you can use import instead';
+        }
+
+        const packageJSONUrl =
+          StringPrototypeStartsWith(this.module.url, 'file://') &&
+            RegExpPrototypeTest(/\.js(\?[^#]*)?(#.*)?$/, this.module.url) &&
+            require('internal/modules/esm/resolve')
+              .getCachedPackageScopeURL(this.module.url);
+        if (packageJSONUrl) {
+          e.message +=
+            `\n'${this.module.url}' cannot be parsed as an ES module, but ` +
+            'it is being treated as an ES module because it has a \'.js\' ' +
+            `file extension and '${packageJSONUrl}' contains "type": ` +
+            '"module". To treat it as a CommonJS script, rename it to use ' +
+            'the \'.cjs\' file extension.';
```

Comment:
I removed the suggestion to modifying the `package.json` because of https://github.com/nodejs/node/pull/37852#discussion_r598428657 and https://github.com/nodejs/node/pull/37852#discussion_r598327318

Reply:
Ok in that case:

> cannot be parsed as an ES module

Do we really need this part, can't we just get straight to `"This file is being treated as an ES module becase..."`?

It would be a nice to have to also have the extension part mentioned for the .mjs case.